packages/block-editor/src/components/use-setting/index.js

```diff
@@ -93,126 +94,158 @@ const removeCustomPrefixes = ( path ) => {
 	return prefixedFlags[ path ] || path;
 };
 
+const mergeCache = new WeakMap();
+function mergeOrigins( value ) {
+	let result = mergeCache.get( value );
+	if ( ! result ) {
+		result = [ 'default', 'theme', 'custom' ].flatMap(
+			( key ) => value[ key ] ?? []
+		);
+		mergeCache.set( value, result );
+	}
+	return result;
+}
+
 /**
  * Hook that retrieves the given setting for the block instance in use.
  *
  * It looks up the settings first in the block instance hierarchy.
  * If none is found, it'll look it up in the block editor store.
```

Comment:
```suggestion
 * If none are found, it'll look them up in the block editor store.
```

Reply:
Pushed a commit that fixes all the typos in the docblocks.